pkg/distributor/query.go

```diff
@@ -232,10 +232,16 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, userID string, re
 					return nil, validation.LimitError(fmt.Sprintf(errMaxChunksPerQueryLimit, util.LabelMatchersToString(matchers), chunksLimit))
 				}
 			}
+
 			for _, series := range resp.Chunkseries {
 				if limitErr := queryLimiter.AddSeries(series.Labels); limitErr != nil {
 					return nil, limitErr
 				}
+				for _, chunks := range series.Chunks {
+					if chunkBytesLimitErr := queryLimiter.AddChunkBytes(chunks.Size()); chunkBytesLimitErr != nil {
```

Comment:
Do we want to divide the size here by the replication factor to handle replicated series on the ingester?

Reply:
I don't think so. The limit is an hard cap on the chunks loaded in memory, so shouldn't be divided by the replication factor.